### PR TITLE
ARROW-2393: [C++] Moving ARROW_CHECK_OK_[PREPEND] macros from status.h into util/logging.h since they use the logging infrastructure and shouldn't be in the public API.

### PR DESCRIPTION
**Pull request by joshuastorck**
Also removing ARROW_RETURN_NOT_OK macro. It is almost exactly the same as RETURN_NOT_OK, but RETURN_NOT_OK is declared inside an arrow namespace block and uses Status instead of ::arrow::Stat. That doesn't really matter, since the macro gets expanded where it's used and users who include Status.h and want to use RETURN_NOT_OK will get compiler errors unless they have a 'using arrow' statement

### CODE REVIEW

**Comment by pitrou** [comment]
We shouldn't remove ARROW_RETURN_NOT_OK, as it's showcased in our docs:
https://arrow.apache.org/docs/cpp/md_tutorials_row_wise_conversion.html
and used in third-party projects:
https://github.com/search?p=&q=ARROW_RETURN_NOT_OK&type=Code

Instead just alias it to RETURN_NOT_OK.

**Comment by pitrou** [comment]
Thanks for the change. Please note there is a linting failure on Travis-CI that needs fixing.

**Review by pitrou — APPROVED** [review]
This looks fine to me. I'll let others chime in before merging.

**Review by xhochy — APPROVED** [review]
+1, LGTM

**Pull request closed without merging**
